doc/rst/technotes/ioSerializers.rst

```diff
@@ -0,0 +1,375 @@
+
+   proc Deserializer.deserializeValue(reader: fileReader,
+                                      ref val: ?readType) : void throws
+
+The ``deserializeType`` method is responsible for creating a new instance of
+the given type, and returning that new instance. By convention
+``deserializeType`` will invoke an initializer by passing in the ``reader``
+and a Deserializer. If a suitable initializer is not available, this method
+may attempt to invoke a ``deserialize`` method on a default-initialized value.
+
+The ``deserializeValue`` method must modify an existing value, which can be
+useful for types that are not cheap to allocate and benefit from re-use (e.g.
+arrays). By convention ``deserializeValue`` will invoke a ``deserialize``
+method on records and classes. If a suitable ``deserialize`` method is not
+available, this method may attempt to invoke a suitable initializer and assign
+the result into the value.
+
+In both methods, the given ``fileReader`` is guaranteed to have a
+``deserializerType`` identical to the type whose method was called. The
+``fileReader`` is also defined to be non-locking.
+
+Note that while both methods may invoke initializers or methods that pass
+control back to the user, Deserializers may ignore those options in the case
+that a class is nilable and can be read as ``nil``.
+
+The Deserializing Initializer
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An initializer invoked by a Deserializer must have the following signature,
+including the argument names "reader" and "deserializer":
+
+.. code-block:: chapel
+
+   proc T.init(reader: fileReader(?),
+               ref deserializer: reader.deserializerType) throws
+
+By default, the compiler will generate a suitable initializer with this
+signature provided that no other user-defined initializers exist.
+
+The ``reader`` and ``deserializer`` are passed separately to help distinguish
+the method signature from other possible initializers, as well as to make it
+slightly more convenient to call methods on the Deserializer. A future release
+will standardize other methods on a Deserializer that provide ways to
+deserialize into common types, like lists or maps.
+
+As with the ``serialize`` method, it is an error for ``reader.deserializer`` to
+refer to a Deserializer other than the ``deserializer`` argument.
+
+Generic types have a slightly more complex initializer signature, in that there
+must be a ``type`` or ``param`` argument for each ``type`` or ``param`` field.
+For example:
+
+.. code-block:: chapel
+
+   record G {
+     type A;
+     type B;
+     var x : A;
+     var y : B;
+   }
+   
+   proc G.init(type A, type B,
+               reader: fileReader, ref deserializer) throws {
+     /* ... */
+   }
+   
+   // With a reader 'r'
+   var x = r.read(G(int, real));
+   // becomes something like...
+   // new G(A=int, B=real, reader=r, deserializer=r.deserializer)
+
+.. warning::
+
+   Generic types with fields like "var x;" do not yet have support for these
```

Comment:
Maybe to make this more specific: "Generic fields that do not have named types, like `var x;`, cannot yet be deserialized using an initializer"